rocky/reports/report_types/ipv6_report/report.py

```diff
@@ -50,12 +50,31 @@ def generate_data(self, input_ooi: str, valid_time: datetime) -> dict[str, Any]:
         for hostname in hostnames:
             if ooi.reference.class_type == IPAddressV6:
                 return {hostname.name: {"enabled": True} for hostname in hostnames}
-            path = Path.parse("Hostname.<hostname [is ResolvedHostname].address")
-            ips = self.octopoes_api_connector.query(path=path, source=hostname.reference, valid_time=valid_time)
+            ips = self.octopoes_api_connector.query(
+                "Hostname.<hostname [is ResolvedHostname].address", valid_time, hostname.reference
+            )
 
             results = {
                 hostname.name: {"enabled": any(ip.reference.class_type == IPAddressV6 for ip in ips)}
                 for hostname in hostnames
             }
 
         return results
+
```

Comment:
Yeah I wasn't sure partly because I knew others were also working on these methods. I think this avoids some merge conflicts, but not sure if I there are any other usages.

Reply:
It avoids conflicts in git, but we still have a conflict when the generate_data is changed but the collect_data is not updated to reflet to changes, so I'm not really sure it is better. We can also remove the old code and check that the new code does the same as the old code in a PR that immediately follows this one, but I don't think it is a good idea to keep maintaining two implementations of the reports.

Reply:
Done